src/lib.rs

```diff
@@ -56,6 +59,11 @@ pub fn hash_to_curve(p: G1Projective) -> G1Projective {
     G1Projective::hash_to_curve(&p.to_compressed(), DOMAIN, &[])
 }
 
+/// returns KeyImage for the given derived public key
```

Comment:
it should probably just be "public key", not "derived public key", since it is kinda meaningless at this level.